vendor/github.com/OpenBazaar/go-ethwallet/wallet/wallet.go

```diff
@@ -472,7 +472,7 @@ func (wallet *EthereumWallet) Spend(amount int64, addr btcutil.Address, feeLevel
 			// but valid txn like some contract condition causing revert
 			if rcpt.Status > 0 {
 				// all good to update order state
-				go wallet.callListeners(wallet.createTxnCallback(hash.Hex(), referenceID, amount, time.Now()))
+				go wallet.AssociateTransactionWithOrder(wallet.createTxnCallback(hash.Hex(), referenceID, amount, time.Now()))
```

Comment:
I appreciate the `wallet.createTxnCallback` helper. I see in every invocation that the timestamp is always `time.Now()`. If you don't anticipate manually setting this to another value, you might internalize this detail instead of asking for it as a param.

Reply:
As of now, I have set it to be time.Now(). However, this can change later if need be to actual block time.